helpers.sh

```diff
@@ -73,3 +73,28 @@ function repo_sync {
 		cp .repo/manifest.xml $archive/manifest.xml
 	fi
 }
+
+function set_base_lmp_version {
+	# Detect the base LmP version we are building on
+	# A user may have place arbitrary tags into their lmp-manifest and we
+	# don't want to accidentally pick one of them up. Git doesn't really
+	# distinguish where tags come from, so:
+
+	# 1: Delete all tags
+	git --git-dir .repo/manifests.git tag | xargs git --git-dir .repo/manifests.git tag -d > /dev/null
+
+	# 2: Get the LMP tags
+	run git --git-dir .repo/manifests.git remote add upstream https://github.com/foundriesio/lmp-manifest
+	run git --git-dir .repo/manifests.git fetch --tags upstream
+
+	# 3: Find our base LMP version based on the HEAD
+	export LMP_VER=$(git --git-dir .repo/manifests.git describe HEAD --tags --abbrev=0)
+	if [[ "${H_PROJECT}" == "lmp" ]] ; then
+		# Public LmP build - we are building for the *next* release
+		LMP_VER=$(( $LMP_VER + 1 ))
+	fi
+
+	# 4: Restore orginal tags - the build may need them
+	run git --git-dir .repo/manifests.git fetch --tags
```

Comment:
At this stage do we need to drop the upstream remote and the tags associated with it? Thinking on possible tag conflicts and what this command would do in case we have conflicts.

I don't think it is something we should be worried with, but thinking on unexpected side effects.

Reply:
can't hurt. I've force pushed this idea and am testing in my factory

Reply:
its working. i think i can merge now if you are fine with this. then i can manually fix some old factories to point at the correct version

Reply:
lgtm